### PR TITLE
New gridworld

### PR DESCRIPTION
**Pull request by zsunberg**
This includes visualization, so waiting on JuliaPOMDP/POMDPModelTools#7 and https://github.com/GiovineItalia/Compose.jl/issues/297

Maybe I will submit another PR without the visualization first.

### CODE REVIEW

**Comment by zsunberg** [comment]
Compose visualization for SimpleGridWorld is supported via Requires.jl for now.

Any comments @MaximeBouton , @rejuvyesh ? I'll probably merge this tonight

File: src/gridworld.jl [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+
+
+# Transitions
+
+POMDPs.isterminal(m::SimpleGridWorld, s::AbstractVector{Int}) = any(s.<0)
+
+
+function POMDPs.transition(mdp::SimpleGridWorld, s::AbstractVector{Int}, a::Symbol)
+    if isterminal(mdp, s) || s in mdp.terminate_in
```
Comment (rejuvyesh) [inline src/gridworld.jl]: Hmm. Shouldn't `isterminal` handle both these cases?

File: src/gridworld.jl [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+POMDPs.reward(mdp::SimpleGridWorld, s::AbstractVector{Int}) = get(mdp.rewards, s, 0.0)
+POMDPs.reward(mdp::SimpleGridWorld, s::AbstractVector{Int}, a::Symbol) = reward(mdp, s)
+
+
+# discount
+
+POMDPs.discount(mdp::SimpleGridWorld) = mdp.discount
+
+# Conversion
+function POMDPs.convert_a(::Type{V}, a::Symbol, m::SimpleGridWorld) where V<:AbstractArray
```
Comment (rejuvyesh) [inline src/gridworld.jl]: `{V<:AbstractArray}`

File: src/gridworld.jl (hunk 1 above)
Comment (zsunberg) [reply src/gridworld.jl]: terminate_in is the set of states that transition to a terminal state *on the next timestep*

Comment (zsunberg) [reply src/gridworld.jl]: could make that clearer, maybe it should be called `terminate_from`

Comment (zsunberg) [reply src/gridworld.jl]: or switching the order ... or maybe I should just add docs :P

File: src/gridworld.jl [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+
+# Actions
+
+POMDPs.actions(mdp::SimpleGridWorld) = (:up, :down, :left, :right)
+Base.rand(rng::AbstractRNG, t::NTuple{L,Symbol}) where L = t[rand(rng, 1:length(t))] # don't know why this doesn't work out of the box
+POMDPs.n_actions(mdp::SimpleGridWorld) = 4
+
+const dir = Dict(:up=>GWPos(0,1), :down=>GWPos(0,-1), :left=>GWPos(-1,0), :right=>GWPos(1,0))
+const aind = Dict(:up=>1, :down=>2, :left=>3, :right=>4)
```
Comment (MaximeBouton) [inline src/gridworld.jl]: minor style comment, I think `direction` instead of `dir`? 
I like all caps name for constant but this is personal maybe.

**Comment by MaximeBouton** [comment]
Looks very nice! thanks

File: src/gridworld.jl (hunk 3 above)
Comment (zsunberg) [reply src/gridworld.jl]: Shoot @MaximeBouton  didn't mean to ignore this!! I think direction is better. I just forgot

**Pull request merged**